### PR TITLE
API-11217 Updated allergy intolerance resource to search with lastUpdated.

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**

### CODE REVIEW

**Comment by jblefkove-va** [comment]
remove dstu2

**Review by bcastillomartel-va — APPROVED** [review]

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/R4AllergyIntoleranceControllerTest.java [hunk 1]
```diff
@@ -34,11 +33,6 @@ public class R4AllergyIntoleranceControllerTest {
 
   AllergyIntoleranceRepository repository = mock(AllergyIntoleranceRepository.class);
 
-  @SneakyThrows
-  static DatamartAllergyIntolerance toDatamart(String json) {
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/R4AllergyIntoleranceControllerTest.java:38]: 🔥

File: data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/r4/AllergyIntoleranceIT.java [hunk 2]
```diff
@@ -47,6 +47,12 @@ public void search() {
             b -> isNotEmpty(b),
             "AllergyIntolerance?identifier={id}",
             testIds.allergyIntolerance()),
+        test(
+            200,
+            AllergyIntolerance.Bundle.class,
+            "AllergyIntolerance?patient={patient}&_lastUpdated={lastUpdated}",
```
Comment (jblefkove-va) [inline data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/r4/AllergyIntoleranceIT.java:53]: This test should also verify that it gets a non-empty bundle

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/R4AllergyIntoleranceTransformer.java [hunk 3]
```diff
@@ -143,6 +145,14 @@ static String verificationStatusCode(DatamartAllergyIntolerance.VerificationStat
     return status.toString();
   }
 
+  private Meta meta(Instant dmLastUpdated) {
+    String lastUpdated = asDateTimeString(dmLastUpdated);
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/R4AllergyIntoleranceTransformer.java:149]: var

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**